src/System.Text.RegularExpressions/tests/Regex.Replace.Tests.cs

```diff
@@ -224,6 +224,31 @@ public void Replace_NoMatch()
             Assert.Same(input, Regex.Replace(input, "no-match", new MatchEvaluator(MatchEvaluator1)));
         }
 
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.RightToLeft)]
+        public void Replace_MatchEvaluatorReturnsNullOrEmpty(RegexOptions options)
+        {
+            string input = "abcde";
+            string result = Regex.Replace(input, @"[abcd]", (Match match) => {
+                switch(match.Value)
+                {
+                    case "a":
+                        return "x";
+                    case "b":
+                        return null;
+                    case "c":
+                        return "";
+                    case "d":
+                        return "y";
+                    default:
+                        throw new InvalidOperationException();
+                }
```

Comment:
If you wanted to be fancy with C# 8, you could use a switch expression :)
```C#
(Match match) => match.Value switch
{
    "a" => "x",
    "b" => null,
    "c" => "",
    "d" => "y",
    _ => throw new InvalidOperationException()
}
```

Reply:
Ooo, I'll definitely do that!